backend/tests/integration/tests/test_compat+separate.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2024 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import pytest
+import time
+
+from testutils.api.client import ApiClient
+from testutils.infra.cli import CliUseradm, CliDeviceauth
+import testutils.api.deviceauth as deviceauth
+import testutils.api.useradm as useradm
+import testutils.api.deployments as deployments
+import testutils.api.inventory as inventory
+
+from testutils.common import (
+    clean_mongo,
+    create_user,
+    mongo,
+)
+
+
+@pytest.fixture(scope="function")
+def user(clean_mongo):
+    deviceauth_cli = CliDeviceauth()
+    useradm_cli = CliUseradm()
+
+    deviceauth_cli.migrate()
+    useradm_cli.migrate()
+    yield create_user("user-foo@acme.com", "correcthorse")
+
+
+class TestClientCompat:
+    def test_compat(self, user):
+        expected_client_versions = [
+            "3.5.3",
+            "4.0.6",
+            "3.0.2",
+            "3.4.0",
+            "3.1.1",
+            "2.3.2",
+            "2.4.2",
+            "3.5.0",
+            "2.6.1",
+            "3.2.1",
+            "2.0.1",
+            "2.5.4",
+            "2.2.1",
+            "2.1.3",
+            "3.3.2",
```

Comment:
There are versions here that are not present in the composition: for instance: 4.0.6

Reply:
yeah this is tricky, this one is hidded in the docker client tag 3.7